runtime/reconcilers/project_parser.go

```diff
@@ -173,16 +174,25 @@ func (r *ProjectParserReconciler) Reconcile(ctx context.Context, n *runtimev1.Re
 	err = repo.Watch(ctx, func(events []drivers.WatchEvent) {
 		// Get changed paths that are not directories
 		changedPaths := make([]string, 0, len(events))
+		deletedPaths := make([]string, 0)
 		for _, e := range events {
 			if e.Dir {
 				continue
 			}
 			if parser.IsIgnored(e.Path) {
+				if !strings.HasPrefix(e.Path, "/tmp") && e.Type == runtimev1.FileEvent_FILE_EVENT_DELETE {
+					// only check for deleted folders that are not inside /tmp
+					deletedPaths = append(deletedPaths, parser.FindFilesInDir(e.Path)...)
+				}
 				continue
 			}
 			changedPaths = append(changedPaths, e.Path)
 		}
 
+		// prepend deleted paths and dedupe
+		deletedPaths = append(deletedPaths, changedPaths...)
+		changedPaths = arrayutil.Dedupe(deletedPaths)
```

Comment:
If we anyway do a dedupe, why not just remove `deletedPaths`, append directly to `changedPaths`, and do a dedupe on it? (As an optimization, would make sense to have a `bool` for whether deleted paths were added to avoid a dedupe in regular cases.)

Reply:
There were previous iterations where it made sense to have a separate list. Will remove it